livre/Eq/Config.cpp

```diff
@@ -248,91 +278,51 @@ void Config::resetCamera( )
+    const ApplicationParameters& params = getApplicationParameters();
+    frameSettings->setFrameNumber( params.frames.x( ));
+
     _impl->framedata.registerObjects();
 
-    if( !registerFrameData_( ))
+    if( !_registerFrameData( ))
         return false;
 
-    if( !eq::Config::init( _impl->framedata.getID( ) ) )
+    if( !eq::Config::init( _impl->framedata.getID( )))
     {
         _impl->framedata.deregisterObjects();
-        deregisterFrameData_();
+        _deregisterFrameData();
         return false;
     }
 
     const eq::Canvases& canvases = getCanvases();
-    if( canvases.empty( ))
-        _impl->currentCanvas = 0;
-    else
-        _impl->currentCanvas = canvases.front();
-
-#ifdef LIVRE_USE_ZEQ
-    if( !servus::Servus::isAvailable( ))
-        return true;
-
-    SubscriberPtr subscriber( new zeq::Subscriber( lunchbox::URI( "hbp://" )));
-    _impl->subscribers.push_back( subscriber );
-    if( _impl->framedata.getAppParameters()->syncCamera )
-    {
-        subscriber->registerHandler( zeq::hbp::EVENT_CAMERA,
-                                     boost::bind( &detail::Config::onHBPCamera,
-                                                 _impl, _1 ));
-    }
-    subscriber->registerHandler( zeq::hbp::EVENT_LOOKUPTABLE1D,
-                                 boost::bind( &detail::Config::onLookupTable1D,
-                                              _impl, _1 ));
-
-    if( !_impl->framedata.getRESTParameters()->useRESTBridge )
-        return true;
-
-#ifdef LIVRE_USE_RESTBRIDGE
-    const std::string subscriberSchema =
-        _impl->framedata.getRESTParameters()->zeqSchema +
-        SUBSCRIBER_SCHEMA_SUFFIX;
-#else
-    const std::string subscriberSchema = "vwscmd://";
-#endif
-    SubscriberPtr vwsSubscriber(
-                new zeq::Subscriber( lunchbox::URI( subscriberSchema )));
-    _impl->subscribers.push_back( vwsSubscriber );
-    vwsSubscriber->registerHandler( zeq::hbp::EVENT_CAMERA,
-                                    boost::bind( &detail::Config::onCamera,
-                                                 _impl, _1 ));
-    vwsSubscriber->registerHandler( zeq::vocabulary::EVENT_REQUEST,
-                                    boost::bind( &detail::Config::onRequest,
-                                                 _impl, _1 ));
-#endif
+    _impl->currentCanvas = canvases.empty() ? 0 : canvases.front();
     return true;
 }
 
-uint32_t Config::startFrame( )
+uint32_t Config::startFrame()
 {
+    FrameSettingsPtr frameSettings = _impl->framedata.getFrameSettings();
+    const ApplicationParameters& params = getApplicationParameters();
+    const uint32_t start = params.frames.x();
+    const uint32_t current = frameSettings->getFrameNumber();
+    const uint32_t end = params.frames.y();
+    const int32_t delta = params.animation;
+    // avoid overflow condition:
+    const uint32_t interval = (end-start+1) == 0 ? 0XFFFFFFFFu : (end-start+1);
```

Comment:
I vote for setting the frame number to 0/undefined (whatever is needed to indicate non-animation) if no interval was specified, rather than relying on the modulo operation doing what you expect. Moreover, is there a check somewhere (latest here) that end >= start?